### DISCUSSION TITLE
Adjustable cron timer?

### DISCUSSION THREAD

**Discussion by k4my4b** [nextcloud/all-in-one#2272] [Ideas] [answered]
I think it would be great to add an environment variable to adjust the built-in cron timer (currently 5 minutes),
something like `NC_CRON_TIMER_MINUTES=2` 
```
#!/bin/bash
set -eu

while true; do
    php -f /var/www/html/cron.php &
    sleep "$NC_CRON_TIMER_MINUTES"m
done
```

**Comment by szaimen** [comment]
Hi, why would you need such a possibility?

**Reply by k4my4b** [reply]
In my case I subscribe to quite a few different calendars which I use with the appointments app and from what I can tell canlenders are pulled/refreshed as a background task which runs once every 5 minutes and 5 minutes a bit too long of a delay between action and seeing the results especially since the appointments app checks against those calendars for conflicts.
I'm not familiar with the inner workings of nextcloud so I cannot say what the ramifications are.

**Comment by szaimen** [comment, marked as answer]
This rather sounds like a workaround to a problem within the appointments app. I would not recommend to go this way and rather tackle the underlying issue in the app itself.

**Reply by k4my4b** [reply]
Truly, it is overkill but the problem is with the calendar app because it uses background tasks to update every so often. The appointments app just checks against whatever data calendar app gives it once the event is fired (i.e. appointment request), essentially trusting that the information is up to date.

**Reply by szaimen** [reply]
Then this needs to be fixed in the calendar app
